build/DownloadCFamilyPlugin/Common.cs

```diff
@@ -0,0 +1,239 @@
+            // open source libraries, both licensed under the MIT license.
+            Common.EnsureWorkingDirectoryExist(destinationFolder, logger);
+
+            var uncompresssedFile = Common.DecompressXZFile(tarFilePath, destinationFolder, logger);
+            Common.ExtractTarToDirectory(uncompresssedFile, destinationFolder, logger);
+        }
+
+        public static string DecompressXZFile(string sourceFilePath, string destinationDirectory, TaskLoggingHelper logger)
+        {
+            var destFile = Path.Combine(destinationDirectory, Path.GetFileName(sourceFilePath)) + ".uncompressed";
+            if (File.Exists(destFile))
+            {
+                LogMessage($"Uncompressed tar file already exists: {destFile}", logger);
+                return destFile;
+            }
+
+            using (Stream xz = new XZStream(File.OpenRead(sourceFilePath)))
+            using (Stream outputStream = new FileStream(destFile, FileMode.CreateNew))
+            {
+                xz.CopyTo(outputStream);
+            }
+            return destFile;
+        }
+
+        public static void UncompressAndUnzipTgz(string tarFilePath, string destinationFolder, TaskLoggingHelper logger)
+        {
+            // The txz file is compressed using XZ compression and zipped using the tar format.
+            // There is no built-in framework support for these format so we are using two
+            // open source libraries, both licensed under the MIT license.
+            EnsureWorkingDirectoryExist(destinationFolder, logger);
+
+            var uncompresssedFile = DecompressGZipFile(tarFilePath, destinationFolder, logger);
+            ExtractTarToDirectory(uncompresssedFile, destinationFolder, logger);
+        }
+
+        public static string DecompressGZipFile(string sourceFilePath, string destinationDirectory, TaskLoggingHelper logger)
+        {
+            var destFile = Path.Combine(destinationDirectory, Path.GetFileName(sourceFilePath)) + ".uncompressed";
+            if (File.Exists(destFile))
+            {
+                LogMessage($"Uncompressed tar file already exists: {destFile}", logger);
+                return destFile;
+            }
+
+            using (Stream xz = new GZipStream(File.OpenRead(sourceFilePath), CompressionMode.Decompress))
+            using (Stream outputStream = new FileStream(destFile, FileMode.CreateNew))
+            {
+                xz.CopyTo(outputStream);
+            }
+            return destFile;
+        }
+
+        public static void ExtractTarToDirectory(string sourceFilePath, string destinationDirectory, TaskLoggingHelper logger)
+        {
+            if (Directory.GetDirectories(destinationDirectory).Length > 0)
+            {
+                // Unzipping the jar is slow so skip if possible
+                LogMessage($"Skipping unzipping the tar because the destination folder already contains a sub-folder.", logger);
```

Comment:
I see that this is new; why do we care that the destination folder has sub folders?

Reply:
Good spot. Left over from the prototype, when unzipping was being done at runtime.
I'll delete it.